sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4175,6 +4175,12 @@ object SQLConf {
     .checkValues(ErrorMessageFormat.values.map(_.toString))
     .createWithDefault(ErrorMessageFormat.PRETTY.toString)
 
+  val REGEX_ENGINE =
```

Comment:
1. I suggest make the value as `Enumeration`, like 

https://github.com/apache/spark/blob/4c938d62d791742b9f0c6a77b66fc06a90d7c0ad/core/src/main/scala/org/apache/spark/internal/config/History.scala#L241-L251

and we should add `checkValues`.

2. the config key should be `spark.sql.regex.engine`

Reply:
Done